services/api/db/serializers.py

```diff
@@ -0,0 +1,109 @@
+    class Meta:
+        model = User
+        fields = ['id', 'email', 'name']
+
+class UserChangePasswordSerialiser(serializers.Serializer):
+
+    #new password type can be customised here.
+    password = serializers.CharField(max_length=255, style={'input_type' : 'passowrd'}, write_only=True)
+    password2 = serializers.CharField(max_length=255, style={'input_type' : 'passowrd'}, write_only=True)
+    class Meta:
+        fields = ['passowrd', 'password2']
+    
+    def validate(self, attrs):
+        password = attrs.get('password')
+        password2 = attrs.get('password2')
+        user = self.context.get('user')
+        if password != password2:
+            raise serializers.ValidationError("Password and Confirm Password doesn't match")
+        user.set_password(password)
+        user.save()
+        return attrs
+
+    
+class SendPasswordResetEmailSerializer(serializers.Serializer):
+    email = serializers.EmailField(max_length=255)
+    class Meta:
+        fields = ['email']
+
+    def validate(self, attrs):
+        email = attrs.get('email')
+        if User.objects.filter(email=email).exists():
+            user = User.objects.get(email = email)
+            uid = urlsafe_base64_encode(force_bytes(user.id))
+            #print('Encoded UID', uid)
+            token = PasswordResetTokenGenerator().make_token(user)
+            #print('Password Reset Token', token)
+            #Instead of local server, desired domain server could be added here.
+            link = 'http://localhost:3000/api/user/reset/'+uid+'/'+token
+            #print('Password Reset Link', link)
+
+            # Here we could provide send email code.
+            return attrs
+        else:
+            raise serializers.ValidationError('You are not a Registered User')
+    
+class UserPasswordResetSerializer(serializers.Serializer):
+    password = serializers.CharField(max_length=255, style={'input_type':'password'}, write_only=True)
+    password2 = serializers.CharField(max_length=255, style={'input_type':'password'}, write_only=True)
+    class Meta:
+        fields = ['password', 'password2']
+
+    def validate(self, attrs):
+        try:
+            password = attrs.get('password')
+            password2 = attrs.get('password2')
+            uid = self.context.get('uid')
+            token = self.context.get('token')
+            if password != password2:
+                raise serializers.ValidationError("Password and Confirm Password doesn't match")
+            id = smart_str(urlsafe_base64_decode(uid))
+            user = User.objects.get(id=id)
+            if not PasswordResetTokenGenerator().check_token(user, token):
+                raise serializers.ValidationError('Token is not Valid or Expired')
+            user.set_password(password)
+            user.save()
+            return attrs
+        except DjangoUnicodeDecodeError as identifier:
+            PasswordResetTokenGenerator().check_token(user, token)
```

Comment:
Do we need this here again?

Reply:
yes because this is for password change , If user who is already in database will ask to change password so we can't reroute him to create_user method again , so in this view before changing password field we need to validate again.

Reply:
removed this .